torch/csrc/jit/codegen/cuda/iter_visitor.cpp

```diff
@@ -26,9 +28,37 @@ std::vector<Statement*> IterVisitor::next(Val* v) {
   return {};
 }
 
-std::vector<Statement*> IterVisitor::next(Expr* expr) {
+std::vector<Statement*> IterVisitor::next(Expr* expr, bool respect_compute_at) {
   FusionGuard::getCurFusion()->assertInFusion(expr, "Cannot traverse expr, ");
-  return {expr->inputs().begin(), expr->inputs().end()};
+  std::vector<Statement*> next_stmts{expr->inputs().begin(),
+                                     expr->inputs().end()};
+  // TODO: Remove the temporary getenv
+  if (std::getenv("FUSER_SORT_INPUT_EXPRS") && respect_compute_at) {
+    if (expr->output(0)->getValType().value() == ValType::TensorView) {
+      auto out = expr->output(0)->as<const TensorView>();
+      // Move input TVs that are computed at this expression backward
+      // so that they are visited later. If multiple inputs are
+      // computed at, move TVs that are computed at an inner loop nest
+      // further backward.
+      std::sort(
```

Comment:
is std::sort stable? I think this would give deterministic orderings, but just making sure.

Reply:
good point. std::sort is not stable (but std::stable_sort() is)

Reply:
So, do we want to use `std::stable_sort`? Both should be deterministic, so I think both are fine.

Reply:
I think it would only be deterministic if stably sorted no? Otherwise if multiple values have the same computeAt position, couldn't the order change between them?

Reply:
Yes, but that's still deterministic, isn't it?

Reply:
While most implementation likely are, I don't think that std::sort() is guaranteed to be deterministic. If that's a concern here, using std::stable_sort() seems an easy solution.

Reply:
I'm ok to just use `std::stable_sort`, but I'm curious what kind of non-deterministic algorithm is actually used.

Reply:
Unless it's documented (guaranteed) to be deterministic we shouldn't relay on that. Even if all current implementation happen to be deterministic, a future library change can break us.

PS. for a hypothetical source of non-determinism, let's say a std::sort() is implemented as a quick sort using a _random_ pivot choice.

Reply:
OK, fair enough. I'll update it.